### PR TITLE
Add FloatingBaseEstimatorDevice YARP wrapper to the algorithms in FloatingBaseEstimator library

### PR DESCRIPTION
**Pull request by prashanthr05**
This PR adds a YARP device wrapper for the FloatingBaseEstimator library, in its current version implementing an InvariantEKF.

**Remark for reviewers:**
Mostly contains small files for configuration and scopes.
Only two files for the source code are significantly big.

### CODE REVIEW

File: devices/FloatingBaseEstimatorDevice/FloatingBaseEstimatorDevice.ini
Comment (prashanthr05) [inline devices/FloatingBaseEstimatorDevice/FloatingBaseEstimatorDevice.ini:4]: This file can be removed with the latest YARP feature. See https://github.com/robotology/whole-body-estimators/issues/70

**Comment by prashanthr05** [comment]
cc @GiulioRomualdi

File: devices/FloatingBaseEstimatorDevice/FloatingBaseEstimatorDevice.ini
Comment (prashanthr05) [reply devices/FloatingBaseEstimatorDevice/FloatingBaseEstimatorDevice.ini]: Done in https://github.com/prashanthr05/bipedal-locomotion-framework/commit/321d50d55bd80d1e46b3ab66427d2a00977ffc96

Comment (prashanthr05) [reply devices/FloatingBaseEstimatorDevice/FloatingBaseEstimatorDevice.ini:4]: Apparently, the YARP version is not compatible, so we will stick to the previous version for the moment.

Comment (prashanthr05) [reply devices/FloatingBaseEstimatorDevice/FloatingBaseEstimatorDevice.ini:4]: Should we add the following lines for the automatic ini installation to work?

```
set(YARP_FORCE_DYNAMIC_PLUGINS ON)
yarp_configure_plugins_installation(package_name)
```

File: devices/FloatingBaseEstimatorDevice/CMakeLists.txt
Comment (S-Dafarra) [inline devices/FloatingBaseEstimatorDevice/CMakeLists.txt:6]: Is this required?

File: devices/FloatingBaseEstimatorDevice/CMakeLists.txt
Comment (S-Dafarra) [inline devices/FloatingBaseEstimatorDevice/CMakeLists.txt:40]: I wonder whether it is worth having a CMake macro for this, as we did for the libraries.

File: devices/FloatingBaseEstimatorDevice/app/CMakeLists.txt
Comment (S-Dafarra) [inline devices/FloatingBaseEstimatorDevice/app/CMakeLists.txt:24]: We have already a macro for the in files https://github.com/dic-iit/bipedal-locomotion-framework/blob/67b78ab1f6942df37077c038f8c09242d2a09f46/cmake/InstallIniFiles.cmake. We can either extend it, rename it, or reuse part of the functionalities.

**Review by S-Dafarra — COMMENTED** [review]
I still need to check the device code

File: devices/FloatingBaseEstimatorDevice/app/CMakeLists.txt
Comment (prashanthr05) [reply devices/FloatingBaseEstimatorDevice/app/CMakeLists.txt]: Thanks for the heads up! This is fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/130/commits/2b370c9e5fef2f69d9fcb9c2a5ee55028a72ee08

File: devices/FloatingBaseEstimatorDevice/CMakeLists.txt
Comment (prashanthr05) [reply devices/FloatingBaseEstimatorDevice/CMakeLists.txt]: A CMake macro for creating a YARP device is added  in https://github.com/dic-iit/bipedal-locomotion-framework/pull/130/commits/4454ca878d36931fb1cf87acc79b4b95b5113462. This file is modified accordingly.

File: devices/FloatingBaseEstimatorDevice/CMakeLists.txt
Comment (prashanthr05) [reply devices/FloatingBaseEstimatorDevice/CMakeLists.txt]: This is removed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/130/commits/4454ca878d36931fb1cf87acc79b4b95b5113462

File: devices/FloatingBaseEstimatorDevice/src/FloatingBaseEstimatorDevice.cpp
Comment (prashanthr05) [reply devices/FloatingBaseEstimatorDevice/src/FloatingBaseEstimatorDevice.cpp]: Thanks for catching this!!  I remembered addressing this but lost the changes during a rebase.

Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/130/commits/fe323bd315395e6d00aaee756b971823696aeb92.

File: devices/FloatingBaseEstimatorDevice/src/FloatingBaseEstimatorDevice.cpp
Comment (prashanthr05) [reply devices/FloatingBaseEstimatorDevice/src/FloatingBaseEstimatorDevice.cpp]: Yes, thanks for catching it. Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/130/commits/fe323bd315395e6d00aaee756b971823696aeb92.

File: devices/FloatingBaseEstimatorDevice/src/FloatingBaseEstimatorDevice.cpp
Comment (prashanthr05) [reply devices/FloatingBaseEstimatorDevice/src/FloatingBaseEstimatorDevice.cpp]: That was actually required in MATLAB (because a reference is returned). 
This is not needed in C++. Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/130/commits/fe323bd315395e6d00aaee756b971823696aeb92.

File: devices/FloatingBaseEstimatorDevice/app/CMakeLists.txt
Comment (S-Dafarra) [reply devices/FloatingBaseEstimatorDevice/app/CMakeLists.txt]: To be honest, I haven't fully understood the modifications. The lines you deleted were not needed?

File: cmake/AddBipedalLocomotionYARPDevice.cmake
Comment (S-Dafarra) [inline cmake/AddBipedalLocomotionYARPDevice.cmake:14]: I would argue that ``DEFAULT`` is a little hard to guess what it is needed for. As far as I understood, it is used to have the plugin compiled by default or not. Since we have already our logic for that, either we remove this input parameter, or we give it a better name.

File: devices/FloatingBaseEstimatorDevice/app/CMakeLists.txt
Comment (prashanthr05) [reply devices/FloatingBaseEstimatorDevice/app/CMakeLists.txt]: I just deleted the lines for the macro that already exists as you pointed out. This macro is used in the subsequent lines.

File: cmake/AddBipedalLocomotionYARPDevice.cmake
Comment (prashanthr05) [reply cmake/AddBipedalLocomotionYARPDevice.cmake]: I removed it in https://github.com/dic-iit/bipedal-locomotion-framework/pull/130/commits/553b80ad0595da3d0346c3112f718c21d1e85f59

File: devices/FloatingBaseEstimatorDevice/app/CMakeLists.txt
Comment (S-Dafarra) [reply devices/FloatingBaseEstimatorDevice/app/CMakeLists.txt]: Ah I see now, thanks!

**Review by S-Dafarra — APPROVED** [review]

**Comment by S-Dafarra** [comment]
Feel free to push the merge button!

**Comment by prashanthr05** [comment]
> Feel free to push the merge button!

Thanks!

@GiulioRomualdi would you have any comments before I merge?

**Review by GiulioRomualdi — APPROVED** [review]
> @GiulioRomualdi would you have any comments before I merge?

I trust @S-Dafarra. @prashanthr05 please feel free to merge it

**Pull request merged**
